packages/app/src/cli/services/dev/processes/app-logs-polling.ts

```diff
@@ -0,0 +1,74 @@
+import {BaseProcess, DevProcessFunction} from './types.js'
+import {pollAppLogs} from '../../app-logs/poll-app-logs.js'
+import {DeveloperPlatformClient} from '../../../utilities/developer-platform-client.js'
+
+import {createLogsDir} from '@shopify/cli-kit/node/logs'
+
+import {outputDebug} from '@shopify/cli-kit/node/output'
+
+interface SubscribeAndStartPollingOptions {
+  developerPlatformClient: DeveloperPlatformClient
+  appLogsQueryOptions: AppLogsQueryOptions
+}
+
+interface AppLogsQueryOptions {
+  shopIds: [string]
+  apiKey: string
+  token: string
+}
+
+export interface AppLogsSubscribeProcess extends BaseProcess<SubscribeAndStartPollingOptions> {
+  type: 'app-logs-subscribe'
+}
+
+interface Props {
+  developerPlatformClient: DeveloperPlatformClient
+  subscription: {
+    shopIds: [string]
+    apiKey: string
+  }
+}
+
+export async function setupAppLogsPollingProcess({
+  developerPlatformClient,
+  subscription: {shopIds, apiKey},
+}: Props): Promise<AppLogsSubscribeProcess | undefined> {
```

Comment:
In what case can we return `undefined`? It looks like we always return `AppLogsSubscribeProcess`.